src/query/index_task.rs

```diff
@@ -0,0 +1,111 @@
+use crate::cluster::{Cluster};
+use std::sync::Arc;
+use crate::errors::{ErrorKind, Result};
+use crate::{ResultCode};
+
+
+
+
+/// Instantiate an IndexTask instance to query Aerospike database cluster for index creation
+/// status.
+#[derive(Debug, Clone)]
+pub struct IndexTask {
+    cluster: Arc<Cluster>,
+    namespace: String,
+    index_name: String
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum IndexTaskResult {
+    NotFound,
+    InProgress,
+    Complete
+}
+
+static SUCCESS_PATTERN: &'static str = "load_pct=";
+static FAIL_PATTERN_201: &'static str = "FAIL:201";
+static FAIL_PATTERN_203: &'static str = "FAIL:203";
+static DELMITER: &'static str = ";";
+
+impl IndexTask {
+
+
+    // TODO: enforce access of this only to Client
+    /// Initializes IndexTask from client, creation should only be expose to Client
+    pub fn new(cluster: Arc<Cluster>, namespace: String, index_name: String) -> Result<Self> {
+        Ok(IndexTask {
+        	cluster: cluster,
+        	namespace: namespace,
+            index_name: index_name
+        })
```

Comment:
Why are you wrapping the return value in a `Result` when there is no chance of the function failing?

Reply:
The function using that was returning `Result` so I did that to match the type, but you are absolutely right in pointing out that its unneccessary, I am removing it.